argilla/docs/how_to_guides/custom_fields.md

```diff
@@ -0,0 +1,184 @@
+---
+description: Learn how to create custom fields using HTML, CSS, and JavaScript templates in Argilla.
+---
+
+# Custom fields with JavaScript, HTML and CSS templates
+
+This guide demonstrates how to create custom fields in Argilla using HTML, CSS, and JavaScript templates.
+
+!!! info "Main Class"
+
+    ```python
+    rg.CustomField(
+        name="custom",
+        title="Custom",
+        template="<div>{{record.fields.custom.key}}</div>",
+        advanced_mode=False,
+        required=True,
+        description="Field description",
+    )
+    ```
+
+    > Check the [CustomField - Python Reference](../reference/argilla/settings/fields.md#src.argilla.settings._field.CustomField) to see the attributes, arguments, and methods of the `CustomField` class in detail.
+
+## Understanding the Record Object
+
+If you are somewhat familiar with web development and thestructure of resources in Argilla, you might will easiliy be able to grasp the `record` object structure you can access within your custom fields definition.The `record` object is the main object that contains all the information about the resource, like `fields`, `metadata`, etc.
+
+## Basic Mode
+
+When `advanced_mode=False`, you can use the `template` argument to only pass basic HTML and CSS templates. To render information, we rely on the [handlebars syntax engine](https://handlebarsjs.com/). This engine will convert the content inside the brackets `{{}}` to the values of record's field's object within your template. As described in the [Accessing record information](#accessing-record-information) section, you can access the fields of the record by navigating to `{{record.fields.<field_name>}}`. For more complex use cases, handlebars has various [expressions, partials, and helpers](https://handlebarsjs.com/guide/) that you can use to render your data.
```

Comment:
```suggestion
By default, custom fields will use the [handlebars syntax engine](https://handlebarsjs.com/) to render templates with `record` information. This engine will convert the content inside the brackets `{{}}` to the values of record's field's object that you reference within your template. As described in the [Accessing record information](#accessing-record-information) section, you can access the fields of the record by navigating to `{{record.fields.<field_name>}}`. For more complex use cases, handlebars has various [expressions, partials, and helpers](https://handlebarsjs.com/guide/) that you can use to render your data. You can deactivate the `handlebars` engine with the `advanced_mode=False` parameter in `CustomField`, then you will need to define custom javascript to access the record attributes.
```

The last sentence could be in a note box with a link to the section below? Or even just not mentioned here?